### DISCUSSION TITLE
Type Error in Collection when typing query.

### DISCUSSION THREAD

**Discussion by mnlfischer** [firebase/firebase-js-sdk#5143] [Q&A] [answered]
Hello, I'm not if it is a bug or wrong implementation.

When I type the query:
``` typescript
export declare function query<T>(query: Query<T>, ...queryConstraints: QueryConstraint[]): Query<T>;
```
as 
``` typescript
const q = query<{foo: string}>(
  collection(db, 'bar')
);
```
I got the following error
```typescript
Argument of type 'CollectionReference<DocumentData>' is not assignable to parameter of type 'Query<{ foot: string; }>'.
```
This is the collection type declaration:
```typescript
export declare function collection(firestore: FirebaseFirestore, path: string, ...pathSegments: string[]): CollectionReference<DocumentData>;
```
This error occurs in 9.0.0-beta.2 - 9.0.0-beta.6
Any advice how to solves this? I tried to merge my interface with document data without success.
Should I use a converter instead without typing the query?

**Comment by Feiyang1** [comment, marked as answer]
Yes, you should use `withConverter` instead of typing `query`s directly. It is by design, so that you can't type query arbitrarily which may not be safe at runtime.

**Reply by mnlfischer** [reply]
works fine. Thanks!
